bioio/bio_image.py

```diff
@@ -215,7 +247,14 @@ def __init__(
             )
             ReaderClass = self._plugin.metadata.get_reader()
         else:
-            # Init reader
+            # check specific reader image types in a situation where a specified reader
+            # only supports some of the ImageLike types.
+            if not BioImage.check_type(image, reader.__init__.__annotations__["image"]):
```

Comment:
The idea here was to pull the typing of the image parameter from the specified reader and make sure they line up. An example of this would be in the ometiff reader (I think just all the readers)[ here]( https://github.com/bioio-devs/bioio-ome-tiff/blob/78030540ae3834535259ec71a57576d2f23e20fd/bioio_ome_tiff/reader.py#L173,) where we define image as pathlike.

Reply:
> annotations has different recommended ways of accessing it for python 39 and before, vs 3.10 and up. https://docs.python.org/3/howto/annotations.html

I might be missing something here. In looking at these documentations I don't believe that annotations directly apply to what we are doing. This is trying to access the typing of the init of the image whereas the examples seem to be referencing a annotation set internally by the class.

Reply:
> > annotations has different recommended ways of accessing it for python 39 and before, vs 3.10 and up. https://docs.python.org/3/howto/annotations.html
> 
> I might be missing something here. In looking at these documentations I don't believe that annotations directly apply to what we are doing. This is trying to access the typing of the init of the image whereas the examples seem to be referencing a annotation set internally by the class.

I put that link because your code uses `__annotations__`  (it says `reader.__init__.__annotations__` right?) and the Python page I linked says:
```This document is designed to encapsulate the best practices for working with annotations dicts. If you write Python code that examines __annotations__ on Python objects, we encourage you to follow the guidelines described below.```

Reply:
Now I am wondering - I'm not sure we guarantee that all Readers have to have an init function that takes a named parameter called `image`.  Is that guaranteed somewhere?